File: Design-Patterns-Using-Swift/Basics OOPS/UsingRelationship.swift
class A {
    init() {
        print("A created")
    }

    func fun1() {
        print("fun1")
    }
}

class B {
    init() {
        print("B created")
    }

    func fun2() {
        print("fun2 start")
        A().fun1()
        print("fun2 end")
    }
}

let b = B()
b.fun2()

File: Design-Patterns-Using-Kotlin/Basics OOPS/UsingRelationship.kt
class A {
    init {
        println("A created")
    }

    fun fun1() {
        println("fun1")
    }
}

class B {
    init {
        println("B created")
    }

    fun fun2() {
        println("fun2 start")
        A().fun1()
        println("fun2 end")
    }
}

fun main() {
    val b = B()
    b.fun2()
}


/*
B created
fun2 start
A created
fun1
fun2 end
*/
